lib/vagrant/machine.rb

```diff
@@ -616,6 +617,25 @@ def with_ui(ui)
       end
     end
 
+    # Returns the SyncedFolder object associated with this machine.
+    #
+    # @return [Class]
+    def synced_folders
+      return @synced_folders if defined?(@synced_folders)
+      @synced_folders = @config.vm.synced_folders.map { |id, opts|
+        if opts[:type]
```

Comment:
I don't really like this bit. But, if the default mount is not in the Vagrantfile it will get added to the synced folders without a type. When the syncing action happens there is a bit that [sets the synced folder type](https://github.com/hashicorp/vagrant/blob/master/lib/vagrant/action/builtin/mixin_synced_folders.rb#L14) if not already set.